riscv-page-tables/src/page_table.rs

```diff
@@ -1296,3 +1333,65 @@ impl<'a, T: PagingMode> Drop for GuestStageMapper<'a, T> {
         }
     }
 }
+
+/// Defines a type of action an iterator should follow
+enum IteratorAction {
+    Proceed(PageSize, SupervisorPageAddr),
+    Skip(PageSize),
+    End,
+}
+
+/// Generate a supervisor page iterator for the given range
+struct SupervisorPageIter<
+    T: PagingMode,
+    F: FnMut(PageAddr<T::MappedAddressSpace>) -> IteratorAction,
+> {
```

Comment:
nit: given how verbose these types are, i'd use `where` to spell out the trait bounds